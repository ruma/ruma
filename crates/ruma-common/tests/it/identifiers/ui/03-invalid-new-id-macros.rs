fn main() {
    let _ = ruma_common::session_id!("invalid~");
}
