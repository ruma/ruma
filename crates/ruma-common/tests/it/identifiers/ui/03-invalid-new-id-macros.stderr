error[E0080]: evaluation of constant value failed
 --> tests/it/identifiers/ui/03-invalid-new-id-macros.rs:2:13
  |
2 |     let _ = ruma_common::session_id!("invalid~");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the evaluated program panicked at 'Invalid Session ID: contains invalid characters', $DIR/tests/it/identifiers/ui/03-invalid-new-id-macros.rs:2:13
  |
  = note: this error originates in the macro `$crate::panic::panic_2021` which comes from the expansion of the macro `ruma_common::session_id` (in Nightly builds, run with -Z macro-backtrace for more info)

note: erroneous constant encountered
 --> tests/it/identifiers/ui/03-invalid-new-id-macros.rs:2:13
  |
2 |     let _ = ruma_common::session_id!("invalid~");
  |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this note originates in the macro `ruma_common::session_id` (in Nightly builds, run with -Z macro-backtrace for more info)
